src/Dapr.PluggableComponents.Tests/DaprPluggableComponentsServiceBuilderTests.cs

```diff
@@ -11,142 +11,224 @@
 // limitations under the License.
 // ------------------------------------------------------------------------
 
-using System.Net.Sockets;
 using Dapr.PluggableComponents.Components;
 using Dapr.PluggableComponents.Components.StateStore;
-using Grpc.Net.Client;
 using Microsoft.Extensions.DependencyInjection;
 using Moq;
 using Xunit;
+using static Dapr.Proto.Components.V1.QueriableStateStore;
 using static Dapr.Proto.Components.V1.StateStore;
+using static Dapr.Proto.Components.V1.TransactionalStateStore;
 
 namespace Dapr.PluggableComponents;
 
-internal interface IMockStateStore : IStateStore
+public sealed class DaprPluggableComponentsServiceBuilderTests
 {
-}
+    [Fact]
+    public async Task RegisterSingletonStateStore()
+    {
+        var mockStateStore = new Mock<IMockStateStore<Unit>>();
 
-internal sealed class MockStateStore : IMockStateStore
-{
-    private readonly IMockStateStore proxy;
+        using var application = DaprPluggableComponentsApplication.Create();
 
-    public MockStateStore(IMockStateStore proxy)
-    {
-        this.proxy = proxy;
-    }
+        application.Services.AddSingleton(_ => mockStateStore.Object);
 
-    #region IStateStore Members
+        using var socketFixture = new SocketFixture();
 
-    public Task DeleteAsync(StateStoreDeleteRequest request, CancellationToken cancellationToken = default)
-    {
-        return this.proxy.DeleteAsync(request, cancellationToken);
-    }
+        application.RegisterService(
+            socketFixture.ServiceOptions,
+            serviceBuilder =>
+            {
+                serviceBuilder.RegisterStateStore<MockStateStore<Unit>>();
+            });
 
-    public Task<StateStoreGetResponse?> GetAsync(StateStoreGetRequest request, CancellationToken cancellationToken = default)
-    {
-        return this.proxy.GetAsync(request, cancellationToken);
-    }
+        await application.StartAsync();
 
-    public Task InitAsync(MetadataRequest request, CancellationToken cancellationToken = default)
-    {
-        return this.proxy.InitAsync(request, cancellationToken);
+        var client = new StateStoreClient(socketFixture.GrpcChannel);
+
+        var metadataA = new Grpc.Core.Metadata { new Grpc.Core.Metadata.Entry(TestConstants.Metadata.ComponentInstanceId, "A") };
+        var metadataB = new Grpc.Core.Metadata { new Grpc.Core.Metadata.Entry(TestConstants.Metadata.ComponentInstanceId, "B") };
+
+        await client.InitAsync(new Proto.Components.V1.InitRequest { Metadata = new Client.Autogen.Grpc.v1.MetadataRequest() }, metadataA);
+        await client.InitAsync(new Proto.Components.V1.InitRequest { Metadata = new Client.Autogen.Grpc.v1.MetadataRequest() }, metadataB);
```

Comment:
I might need some help: what exactly is the intent/being tested here? The multiple instances behind a single component? Is this dapr/dapr#5928?

Reply:
This test ensures that, when using the "singleton" style of component registration, the same single instance of the component is invoked for every request, even when they are made from separately configured components (i.e. two component YAML files that point to the same socket).  That's as opposed to the following test which ensures that, when using the "multiplexing" style of component registration, that the instance associated with a given component (YAML) is invoked only when *that* specific component invokes a method.

It's related to that Dapr proposal; `x-component-instance` is already being attached to each method invocation; the proposal is to add a new piece of metadata allowing an additional axis of multiplexing, but as mentioned, making `x-component-instance` unique/random enables the same end result using the existing logic.